module-db/Tables/MultimediaFilesTable.cpp

```diff
@@ -386,4 +386,28 @@ namespace db::multimedia_files
 
         return (*retQuery)[0].getUInt32();
     }
+
+    auto MultimediaFilesTable::getOffsetOfSortedRecordByPath(const std::string &folderPath,
+                                                             const std::string &recordPath,
+                                                             SortingBy sorting) -> SortedRecord
+    {
+        const std::string query = "SELECT * FROM ("
+                                  "      SELECT"
+                                  "          ROW_NUMBER () OVER ("
+                                  "              ORDER BY " +
+                                  getSorting(sorting) +
+                                  "          ) offset,"
+                                  "          path"
+                                  "      FROM"
+                                  "          files"
+                                  "      WHERE path LIKE '" +
+                                  folderPath + "%%'" +
+                                  "  )"
+                                  "  WHERE"
+                                  "      path='" +
+                                  recordPath + "'";
+
+        std::unique_ptr<QueryResult> retQuery = db->query(query.c_str());
+        return SortedRecord{.path = (*retQuery)[1].getString(), .offset = (*retQuery)[0].getUInt32()};
```

Comment:
You can omit the type declaration here and just use `{}`, but if you prefer it with explicit type, feel free to resolve :wink:

```suggestion
        return {.path = (*retQuery)[1].getString(), .offset = (*retQuery)[0].getUInt32()};
```

Reply:
IMO it is slightly more readable with the exact type, so I'll leave it as it is :)